src/libbson/src/bson/bson-compat.h

```diff
@@ -171,16 +171,34 @@ typedef signed char bool;
 #endif
 
 
+#if defined(USE_LEGACY_GCC_ATOMICS) || (!defined(__clang__) && \
+   __GNUC__ == 4 && __GNUC_MINOR__ >= 1 && __GNUC_MINOR__ < 7)
+#define BSON_USE_LEGACY_GCC_ATOMICS
+#else
+#undef BSON_USE_LEGACY_GCC_ATOMICS
+#endif
```

Comment:
If this is only applicable to atomics, and not generically useful, I suggest moving `BSON_USE_LEGACY_GCC_ATOMICS` to bson-atomic.h and undefining it.

Reply:
Ibid,